puffin/src/trace.rs

```diff
@@ -337,6 +343,8 @@ pub struct TraceContext<PB: ProtocolBehavior> {
     spawner: Spawner<PB>,
 
     phantom: PhantomData<PB>,
+    /// The number of steps that have been successfully executed
+    pub executed_until: usize,
 }
```

Comment:
hmm, this is weird. A trace context is not necessarily bound to a single trace.

Reply:
Ok, I see `executed_until` should be the number of step executed in the last trace (excluding prior traces). As we discussed, recursive prior traces should be removed in favor of a Vec of traces, but at the moment counting the number of executed steps with potentially recursive prior traces is too much work for no real benefit

Reply:
is it ok ?